scripts/research/lib/research_harness.py

```
@@ -0,0 +1,579 @@
+    repo_manifest_path: Path,
+    dataset_root: Path,
+    identity: DatasetIdentityContract,
+    policy: OutcomeAccessPolicy,
+) -> AuthorizedDataset:
+    """Verify dataset identity before a runner can obtain outcome paths."""
+    try:
+        repo_manifest = yaml.safe_load(repo_manifest_path.read_text(encoding="utf-8"))
+    except FileNotFoundError as exc:
+        raise DatasetIdentityError(
+            f"missing repository dataset manifest: {repo_manifest_path}"
+        ) from exc
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)